src/components/menu/menu.tsx

```diff
@@ -627,6 +615,11 @@ export class Menu {
             const menuLoader = menuItem.items as MenuLoader;
             this.loadingSubItems = true;
             const subItems = await menuLoader(menuItem);
+
+            if (this.selectedMenuItem !== menuItem) {
```

Comment:
Same as the comment above about this probably never going to return true and thus not work as expected, as well as missing the reset of `this.loadingSubItems` to `false`.

Reply:
Right, was missing a reset b9c0f16c9e9af8fde0f342920cc5afe019f25a15

We're setting `this.loadingSubItems` to `false` in `this.clearSearch()` so I think it's covered.

Reply:
c13d5483183ce15226cb6e39e0ccfcb54c9d3141